controllers/cluster_controller.go

```diff
@@ -234,7 +252,7 @@ func (r *ClusterReconciler) reconcile(ctx context.Context, log logr.Logger, clus
 		return reconcileResult.ToCtrlResult(), nil
 	}
 
-	if cluster.IsSelfManaged() {
+	if cluster.IsSelfManaged() && !r.experimentalSelfManagedUpgrade {
```

Comment:
Are you going to still use the existing interface `ReconcileWorkerNodes` when the flag set to true to reconcile worker node?

Reply:
if the flag is set to true, we just call `Reconcile`, which will also reconcile worker nodes aside from CP :)